src/main/java/org/dependencytrack/model/Project.java

```diff
@@ -297,10 +298,9 @@ public enum FetchGroup {
     private Date inactiveSince;
 
     @Persistent(table = "PROJECT_ACCESS_TEAMS", defaultFetchGroup = "true")
-    @Join(column = "PROJECT_ID")
+    @Join(column = "PROJECT_ID", primaryKey = "PROJECT_ACCESS_TEAMS_PK")
     @Element(column = "TEAM_ID")
-    @Order(extensions = @Extension(vendorName = "datanucleus", key = "list-ordering", value = "name ASC"))
-    private List<Team> accessTeams;
+    private Set<Team> accessTeams;
```

Comment:
Making this a `Set` causes DataNucleus to generate a primary key on `PROJECT_ID` and `TEAM_ID`. The PK ensures that neither column can be `null`, and that rows are unique.

We mirror this behavior in the DB changelog, and create a primary key, just like DN would do it.